@xen-orchestra/rest-api/src/groups/group.controller.mts

```diff
@@ -49,4 +69,20 @@ export class GroupController extends XoController<XoGroup> {
   getGroup(@Path() id: string): Promise<Unbrand<XoGroup>> {
     return this.getObject(id as XoGroup['id'])
   }
+
+  /**
+   * @example body {
+   * "name": "new group"
+   * }
+   */
+  @Example(groupId)
+  @Post('')
+  @Middlewares(json())
+  @SuccessResponse(createdResp.status, createdResp.description)
+  @Response(invalidParameters.status, invalidParameters.description)
+  async createGroup(@Body() body: { name: string }): Promise<{ id: Unbrand<XoGroup>['id'] }> {
+    const group = await this.restApi.xoApp.createGroup(body)
```

Comment:
If your group already exists, this function will generate an error, which will be caught by error middleware.
Could you update this middleware to add more information when `objectAlreadyExists` is caught?

```ts
// generic-error-handler.middleware.mts
...
else if(objectAlreadyExists.is...){
   res.status(409)
  responseError.data = (error as XoError).data
}
```

This will make the error returned when `objectAlreadyExists` is caught more user friendly.